File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] arr, bool[] flag) 
    {
        var list = new List<int>();
        
        for(int i = 0; i < flag.Length; ++i)
        {
            int repeat = arr[i];
            if(flag[i])
            {
                for(int k = 0; k < repeat * 2; ++k)
                    list.Add(repeat);
            }
            else
            {
                for(int k = 0; k < repeat; ++k)
                    list.RemoveAt(list.Count - 1);
            }
        }
        
        return list.ToArray();
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<bool> flag) {
    vector<int> answer;
    
    for (int i = 0; i < arr.size(); ++i)
    {
        int n = arr[i];
        if (flag[i])
        {
            answer.insert(answer.end(), n * 2, n);
        }
        else
        {
            answer.erase(answer.end() - n, answer.end());
        }
    }
    
    return answer;
}